### PR TITLE
Fix slight issues with autocompletion and member lists in GDScript

### LINKED ISSUE #27152: Code completion for members/methods inherited from built-in classes does not work properly
**Godot version:** 3.1

**OS/device including version:** Ubuntu 18.04

**Issue description:**
Say, we have the following inheritance tree (`<---` means 'inherits' or 'extends'):
```
Node2D <--- intermediate.gd <--- script.gd <--- another_script.gd
```

Node2D's members and methods are in the completion list only for `intermediate.gd`:
```gdscript
# intermediate.gd
extends Node2D
class_name Intermediate

func _ready():
	# COMPLETION WORKS
	print (position)
	
	# COMPLETION WORKS
	rotate(1)

func custom_func():
	pass
```

Same members and methods ***are not*** listed if you edit scripts that inherit `intermediate.gd`:
```gdscript
# script.gd
extends Intermediate

func _ready():
	# NO CODE COMPLETION FOR INHERITED FIELDS
	print (position)
	
	# COMPLETION DOES NOT WORK
	rotate(1);
	
	# COMPLETION WORKS QUITE WELL
	custom_func()
```
```gdscript
# another_script.gd
extends "res://script.gd"

func _ready():
	# COMPLETION WORKS
	custom_func()
	
	# COMPLETION DOES NOT WORK
	print (position)
	
	# COMPLETION DOES NOT WORK
	rotate(1)
```

The only dirty work-around for this is to simply cast `self` to whatever class you inherit:
```gdscript
# script.gd
# ...
    # why do I have to cast to the class I already inherit to get code completion?
    (self as Node2D).rotate(1)
# ...
```

**Minimal reproduction project:**
[repro_field_completion.tar.gz](https://github.com/godotengine/godot/files/2974859/repro_field_completion.tar.gz)

### LINKED ISSUE #28591: AnimationPlayer call method does not work with extended classes
**Godot version:**

3.1

**OS/device including version:**

Windows 10

**Issue description:** 

When setting up an AnimationPlayer with a call method track on a node that uses a script that extends another script, i.e. using AnimationPlayer to call a method in class B that extends class A, class A methods are not listed, but class B methods are listed twice.

**Steps to reproduce:**
1. Make a class_name A
2. Have a new script (B) extend A
3. Set up a call method track on an AnimationPlayer that calls a method on a node with the B script
4. Insert a key on the call method track and look at the list of functions available

**Minimal reproduction project:**

[AnimationPlayerInheritedClassBug.zip](https://github.com/godotengine/godot/files/3136466/AnimationPlayerInheritedClassBug.zip)

### PR DESCRIPTION
**Pull request by bojidar-bg**
Fixes #27152
Fixes #28591

### CODE REVIEW

File: modules/gdscript/gdscript_editor.cpp [hunk 1]
```diff
@@ -1951,7 +1951,7 @@ static void _find_identifiers_in_base(const GDScriptCompletionContext &p_context
 					} else {
 						base_type.has_type = script->get_instance_base_type() != StringName();
 						base_type.kind = GDScriptParser::DataType::NATIVE;
-						base_type.script_type = script->get_instance_base_type();
```
Comment (bojidar-bg) [inline modules/gdscript/gdscript_editor.cpp]: The glories of static typing.. and auto-coercion.

Note that `get_instance_base_type` returns `StringName`, while `script_type` is a `Ref<Script>`. How C++ manages to convert one into the other is a complete mystery to me. 
Explanations are welcome!

File: modules/gdscript/gdscript.cpp [hunk 2]
```diff
@@ -226,7 +226,7 @@ void GDScript::get_script_method_list(List<MethodInfo> *p_list) const {
 
 	const GDScript *current = this;
 	while (current) {
-		for (const Map<StringName, GDScriptFunction *>::Element *E = member_functions.front(); E; E = E->next()) {
+		for (const Map<StringName, GDScriptFunction *>::Element *E = current->member_functions.front(); E; E = E->next()) {
```
Comment (akien-mga) [inline modules/gdscript/gdscript.cpp]: Looks like a bug that a static analysis tool might have been able to figure out :D

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**